### PR TITLE
Add TypeScript types

### PR DESCRIPTION
**Pull request by karaggeorge**
This includes types implemented in #955 so it should wait until that gets merged before going in

### CODE REVIEW

**Comment by karaggeorge** [comment]
Tests are failing because `tsc` can't run on Node 8. Can I remove v8 from the list? Or should I try using an older `tsc` version? 

Or maybe removing the type test all together

**Comment by sindresorhus** [comment]
> Can I remove v8 from the list?

Done in `main` branch.

File: index.d.ts [hunk 1]
```diff
@@ -0,0 +1,151 @@
...
+    id: string;
+    name: string;
+  }
+
+  type VideoCodec = 'h264' | 'hevc' | 'proRes422' | 'proRes4444'
+
+  type RecordingOptions = {
+    /**
+     * Number of frames per seconds.
+     */
```
Comment (sindresorhus) [inline index.d.ts:18]:
```suggestion
    /**
    Number of frames per seconds.
    */
```

Drop the star vertical border. Makes it too hard to edit the comments. Applies to this whole file.

**Comment by sindresorhus** [comment]
Can you fix the merge conflict and rebase from `main` branch?

File: index.d.ts (hunk 1 above)
Comment (karaggeorge) [reply index.d.ts:18]: Makes sense. I wonder if there's an editor setting to disable it. I'll make the updates today and push

**Pull request merged**